dask/array/enhanced_numpy.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import absolute_import, division, print_function
+
+from functools import wraps
+from inspect import getargspec
+
+import numpy as np
+
+from ..utils import ignoring
+
+
+
+# A wrapper for functions that don't provide keepdims to ensure that they do.
+def keepdims_wrapper(a_callable):
+    if "keepdims" in getargspec(a_callable).args:
+        return a_callable
+
+    @wraps(a_callable)
+    def keepdims_wrapped_callable(x, axis=None, keepdims=None, *args, **kwargs):
+        r = a_callable(x, axis=axis, *args, **kwargs)
+
+        if not keepdims:
+            return r
+
+        axes = axis
+
+        if axes is None:
+            axes = range(x.ndim)
+
+        try:
+            len(axes)
+        except TypeError:
+            axes = [axes]
+
+        r_slice = tuple()
+        for each_axis in range(x.ndim):
+            if each_axis in axes:
+                r_slice += (None,)
+            else:
+                r_slice += (slice(None),)
+
+        r = r[r_slice]
+
+        return r
+
+    return keepdims_wrapped_callable
+
+
+# Wrap NumPy functions to ensure they provide keepdims.
+sum = keepdims_wrapper(np.sum)
+prod = keepdims_wrapper(np.prod)
+min = keepdims_wrapper(np.min)
+max = keepdims_wrapper(np.max)
+argmin = keepdims_wrapper(np.argmin)
+nanargmin = keepdims_wrapper(np.nanargmin)
+argmax = keepdims_wrapper(np.argmax)
+nanargmax = keepdims_wrapper(np.nanargmax)
+any = keepdims_wrapper(np.any)
+all = keepdims_wrapper(np.all)
+nansum = keepdims_wrapper(np.nansum)
+
+with ignoring(AttributeError):
+    nanprod = keepdims_wrapper(np.nanprod)
+
+nanmin = keepdims_wrapper(np.nanmin)
+nanmax = keepdims_wrapper(np.nanmax)
+mean = keepdims_wrapper(np.mean)
+
+with ignoring(AttributeError):
+    nanmean = keepdims_wrapper(np.nanmean)
+
+var = keepdims_wrapper(np.var)
+
+with ignoring(AttributeError):
+    nanvar = keepdims_wrapper(np.nanvar)
+
+std = keepdims_wrapper(np.std)
+
+with ignoring(AttributeError):
+    nanstd = keepdims_wrapper(np.nanstd)
```

Comment:
Yes, I would dump all of this into `chunk.py` which I intended to serve as "a bunch of functions on single numpy chunks that numpy itself doesn't support."  It's a bit more filled out in other branches.  Probably we should find a better name, maybe enhanced_numpy is it.